library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -1357,15 +1365,20 @@ impl<T, A: Allocator> VecDeque<T, A> {
     ///
     /// let mut deque = VecDeque::new();
     ///
-    /// deque.push_back(0);
     /// deque.push_back(1);
+    /// deque.push_back(2);
+    /// deque.push_back(3);
+    /// deque.push_back(4);
     ///
-    /// deque.push_front(10);
-    /// deque.push_front(9);
+    /// let (front, back) = deque.as_mut_slices();
     ///
-    /// deque.as_mut_slices().0[0] = 42;
-    /// deque.as_mut_slices().1[0] = 24;
-    /// assert_eq!(deque.as_slices(), (&[42, 10][..], &[24, 1][..]));
+    /// if !front.is_empty() {
+    ///     front[0] = 5;
+    ///     assert_eq!(deque[0], 5);
+    /// } else {
+    ///     back[back.len() - 1] = 5;
+    ///     assert_eq!(deque[deque.len() - 1], 5);
+    /// }
```

Comment:
Similarly, it may be a bit cleaner to `if` on the indexing and just assert the final result:

```rust
deque.push_back(0);
deque.push_back(1);

deque.push_front(10);
deque.push_front(9);

// Since the split point is not guaranteed, we may need to update
// either slice.
let mut update_nth = |index: usize, val: u32| {
    let (front, back) = deque.as_mut_slices();
    if index > front.len() - 1 {
        back[index - front.len()] = val;
    } else {
        front[index] = val;
    }
};

update_nth(0, 42);
update_nth(2, 24);

let v: Vec<_> = deque.into();
assert_eq!(v, [42, 10, 24, 1]);
```